### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by Dangelo-JAN**
# I create decorators to my ruby app: OOP school library: decorate a class

In this milestone I:

Interface:
- Create a class `Nameable`.
   - Implement a method called `correct_name` that will raise a `NotImplementedError`.

Turn my Person class to Nameable:
- My Person class inherits from Nameable
- A method `correct_name` was implemented. It should simply return the name attribute.

Prepare base Decorator
- Inherits from Nameable.
- In the constructor I assign a nameable object from params to an instance variable.
- Implement the `correct_name` method that returns the result of the correct_name method of the `@nameable`.

Prepare CapitalizeDecorator and TrimmerDecorator:
- For the CapitalizeDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that capitalizes the output of `@nameable.correct_name`.
- For the TrimmerDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that makes sure that the output of `@nameable.correct_name` has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
+require_relative 'nameable'
```
Comment (arslanbisharat) [inline person.rb:1]: - [x] According to the guidelines, please make sure the Person class inherits from Nameable class.

File: capitalize_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require_relative 'decorator'
```
Comment (arslanbisharat) [inline capitalize_decorator.rb:1]: - [x] You have added a readme file but please make sure the Readme File is descriptive enough with details about your project and the authors.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
Hi @Dangelo-JAN,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The Pull Request has a proper title and description. ✔️
- You have added a descriptive Readme file. Good Job. ✔️
- There is no linter error is present. ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Please make sure you right descriptive comments. Commenting involves placing Human Readable Descriptions inside of computer programs detailing what the Code is doing. Proper use of commenting can make code maintenance much easier, as well as helping make finding bugs faster. Further, commenting is very important when writing functions that other people will use. :x:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (Dangelo-JAN) [reply person.rb:1]: done

File: capitalize_decorator.rb (hunk 2 above)
Comment (Dangelo-JAN) [reply capitalize_decorator.rb:1]: done

**Review by Kingstalux — APPROVED** [review]
Hi @Dangelo-JAN,
# Status: Approved ✔️ ✔️ ✔️ 
 
128382030-38e7de91-f6f2-45cf-b54e-8068e0446e4c

Good job implementing the required changes 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**